flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/testutils/BatchCompiledPlanTestBase.java

```diff
@@ -0,0 +1,262 @@
+import org.apache.flink.table.test.program.TableTestProgramRunner;
+import org.apache.flink.table.test.program.TestStep.TestKind;
+import org.apache.flink.test.junit5.MiniClusterExtension;
+import org.apache.flink.types.Row;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.MethodOrderer.OrderAnnotation;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Base class for implementing compiled plan tests for {@link BatchExecNode}. You can generate json
+ * compiled plan for the latest node version by running {@link
+ * BatchCompiledPlanTestBase#generateCompiledPlans(TableTestProgram)}. This method does not recreate
+ * the compiled plan if it already exists for the given version of the operator.
+ *
+ * <p><b>Note:</b> The test base uses {@link TableConfigOptions.CatalogPlanCompilation#SCHEMA}
+ * because it needs to adjust source and sink properties. Therefore, the test base can not be used
+ * for testing storing table options in the compiled plan.
+ */
+@ExtendWith(MiniClusterExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@TestMethodOrder(OrderAnnotation.class)
+public abstract class BatchCompiledPlanTestBase implements TableTestProgramRunner {
+
+    private final Class<? extends ExecNode<?>> execNodeUnderTest;
+    private final List<Class<? extends ExecNode<?>>> childExecNodesUnderTest;
+
+    protected BatchCompiledPlanTestBase(Class<? extends ExecNode<?>> execNodeUnderTest) {
+        this(execNodeUnderTest, new ArrayList<>());
+    }
+
+    protected BatchCompiledPlanTestBase(
+            Class<? extends ExecNode<?>> execNodeUnderTest,
+            List<Class<? extends ExecNode<?>>> childExecNodesUnderTest) {
+        this.execNodeUnderTest = execNodeUnderTest;
+        this.childExecNodesUnderTest = childExecNodesUnderTest;
+    }
+
+    // Used for testing Restore Test Completeness
+    public Class<? extends ExecNode<?>> getExecNode() {
+        return execNodeUnderTest;
+    }
+
+    // Used for testing Restore Test Completeness
+    public List<Class<? extends ExecNode<?>>> getChildExecNodes() {
+        return childExecNodesUnderTest;
+    }
+
+    @Override
+    public EnumSet<TestKind> supportedSetupSteps() {
+        return EnumSet.of(
+                TestKind.CONFIG,
+                TestKind.FUNCTION,
+                TestKind.TEMPORAL_FUNCTION,
```

Comment:
Temporal functions are not supported in batch mode.

Reply:
Removing. 

Should this be "TEMPORARY_FUNCTION" rather than TEMPORAL?

Looking at the usage, it appears to be setting up temporary functions?

Reply:
No temporal functions are another category of deprecated functions.